Spatial.gd

```diff
@@ -413,13 +420,81 @@ func gen_map_markers():
+			#Some icons have their texture in a parent of the category.
 			continue
 		var full_texture_path = self.marker_file_dir + texture_path.get_path()
-		gen_new_icon(position_vector, full_texture_path)
+		gen_new_icon(position_vector, full_texture_path, icon.get_category().get_name())
+	
+		
+onready var local_data = get_node("Control/LocalData") 
+
+func build_category_tree():
+	var root = marker_packs.create_item()
+	root.set_text(0, "Markers available on current map")
+	root.set_selectable(0, false)
+	root.set_text(1, "Visible")
+	
+	for category in self.markerdata.get_category():
+		add_category(root, category, category.get_name(), false)
+
+
+func add_category(item: TreeItem, category, tree_path: String, collapsed: bool):
+	if category.get_name() != "":
+		var category_item = marker_packs.create_item(item)
+		category_item.set_text(0,category.get_display_name())
+		category_item.set_metadata(0,category)
+		category_item.set_cell_mode(1, TreeItem.CELL_MODE_CHECK)
+		category_item.set_checked(1, self.local_data.visible_dict.get(tree_path, false))
+		category_item.set_tooltip(1, "Show/Hide")
+		category_item.set_editable(1, true)
+		category_item.set_collapsed(collapsed)
+		for category_child in category.get_children():
+			add_category(category_item, category_child, tree_path + "." + category_child.get_name(), true)
+
+
+func switch_selected_category(category_item: TreeItem):
+	var selected_category_name = find_pedigree_name(category_item, category_item.get_metadata(0).get_name())
+	local_data.visible_dict[selected_category_name] = category_item.is_checked(1)
+	switch_checkboxes(category_item, category_item.is_checked(1))
+	
+	node_checkmark_switch(selected_category_name, category_item, self.paths)
+	node_checkmark_switch(selected_category_name, category_item, self.minimap)
+	node_checkmark_switch(selected_category_name, category_item, self.icons)
+
+	$Control/LocalData.save_local()
+
+
+func find_pedigree_name(category_item: TreeItem, pedigree_name: String):
+	var parent =  category_item.get_parent()
+	if parent.is_selectable(0):
+		pedigree_name = parent.get_metadata(0).get_name() + "." + pedigree_name
+		pedigree_name = find_pedigree_name(parent, pedigree_name)
+	return pedigree_name
+
+
+func node_checkmark_switch(selected_category_name: String, category_item, nodes):
+	for node in nodes.get_children():
+		if node.category_name.begins_with(selected_category_name):
+			local_data.visible_dict[node.category_name] = category_item.is_checked(1)
+			if category_item.is_checked(1):
+				node.show()
+			else:
+				node.hide()
+
+
+func switch_checkboxes(item: TreeItem, checked: bool):
+	if item.get_cell_mode(1) == TreeItem.CELL_MODE_CHECK:
+		item.set_checked(1, checked)
+		local_data.visible_dict[find_pedigree_name(item, item.get_metadata(0).get_name())] = checked
+		
+	var child_item = item.get_children()
+	while child_item != null:
+		switch_checkboxes(child_item, checked)
+		child_item = child_item.get_next()
```

Comment:
As discussed offline this might not be necessary with the user experience changes.